client/web/src/insights/pages/creation/search-insight/utils/use-url-query-insight/use-url-query-insight.ts

```diff
@@ -66,32 +54,66 @@ export function getInsightDataFromQuery(query: string | null): InsightData | nul
 
     return {
         seriesQuery: dedupeWhitespace(humanReadableQueryString),
-        repositories: getSanitizedRepositoriesString(repositories),
+        repositories,
     }
 }
 
-/**
- * Returns initial values for the search insight from query param 'insight-query'.
- */
-export const getUrlQueryInsight = memoize((queryParameters: string): CreateInsightFormFields | null => {
-    const queryParametersString = new URLSearchParams(queryParameters).get('query')
-
-    const insightData = getInsightDataFromQuery(queryParametersString)
-
-    if (insightData === null) {
-        return null
-    }
-
+function createInsightFormFields(seriesQuery: string, repositories: string[] = []): CreateInsightFormFields {
     return {
         ...INITIAL_INSIGHT_VALUES,
         series: [
             createDefaultEditSeries({
                 edit: true,
                 valid: true,
                 name: 'Search series #1',
-                query: insightData.seriesQuery,
+                query: seriesQuery ?? '',
             }),
         ],
-        repositories: insightData.repositories,
+        repositories: repositories.join(', '),
     }
-})
+}
+
+export interface UseURLQueryInsightResult {
+    /**
+     * Insight data. undefined in case if we are in a loading state or
+     * URL doesn't have query param.
+     * */
+    data: CreateInsightFormFields | ErrorLike | undefined
+
+    /** Whether the search query  param is presented in URL. */
+    hasQueryInsight: boolean
+}
+
+/**
+ * Returns initial values for the search insight from query param.
+ */
+export function useURLQueryInsight(queryParameters: string): UseURLQueryInsightResult {
+    const { getResolvedSearchRepositories } = useContext(InsightsApiContext)
+    const [insight, setInsight] = useState<CreateInsightFormFields | ErrorLike | undefined>()
```

Comment:
Having a `setInsight` name here is a bit confusing. Can we signify that this state contains `FormFields`?

```suggestion
    const [insightFormFields, setInsightFormFields] = useState<CreateInsightFormFields | ErrorLike | undefined>()
```